tests/path_tracing/test_path_tracing_midpoint.py

```diff
@@ -0,0 +1,422 @@
+            ptf_port_idx, pc_ports_map, mg_facts["minigraph_ptf_indices"], ingress_router_mac
+
+        for prefix in ["2001:db8:2::1/128"]:
+            duthost.shell(duthost.get_vtysh_cmd_for_namespace(
+                "vtysh -c \"configure terminal\" -c \"no ip route {} {} tag 1\""
+                .format(prefix, selected_peer_ip_ifaces_pairs[1][0]), ptf_port_idx_namespace))
+
+    def teardown_path_tracing(self, duthost):
+        """
+        teardown Path Tracing after test by disabling Path Tracing on all interfaces
+        :param duthost: DUT host object
+        :return:
+        """
+        logger.info("Disable Path Tracing")
+        self.config_path_tracing(duthost, ifname="Ethernet8", enable=False)
+
+    def simple_ipv6_packet(
+        pktlen=100,
+        eth_dst="00:01:02:03:04:05",
+        eth_src="00:06:07:08:09:0a",
+        ipv6_src="2001:db8:85a3::8a2e:370:7334",
+        ipv6_dst="2001:db8:85a3::8a2e:370:7335",
+        ipv6_tc=0,
+        ipv6_ecn=None,
+        ipv6_dscp=None,
+        ipv6_hlim=64,
+        ipv6_fl=0,
+    ):
+        """
+        Return a simple IPv6 packet
+
+        Supports a few parameters:
+        @param len Length of packet in bytes w/o CRC
+        @param eth_dst Destination MAC
+        @param eth_src Source MAC
+        @param ipv6_src IPv6 source
+        @param ipv6_dst IPv6 destination
+        @param ipv6_tc IPv6 traffic class
+        @param ipv6_ecn IPv6 traffic class ECN
+        @param ipv6_dscp IPv6 traffic class DSCP
+        @param ipv6_ttl IPv6 hop limit
+        @param ipv6_fl IPv6 flow label
+
+        Generates a simple IPv6 packet. Users shouldn't assume anything about this
+        packet other than that it is a valid ethernet/IPv6 frame.
+        """
+
+        ipv6_tc = testutils.ip_make_tos(ipv6_tc, ipv6_ecn, ipv6_dscp)
+
+        pkt = packet.Ether(dst=eth_dst, src=eth_src)
+        pkt /= packet.IPv6(
+            src=ipv6_src, dst=ipv6_dst, fl=ipv6_fl, tc=ipv6_tc, hlim=ipv6_hlim
+        )
+
+        return pkt
+
+    def ipv6_packet_no_hbh_pt(self, common_param, ptfadapter):
+        """ create IPv6 packet for testing """
+
+        (peer_ip_ifaces_pair, rif_rx_ifaces, rif_support, ptf_port_idx,
+         pc_ports_map, ptf_indices, ingress_router_mac) = common_param
+
+        return TestPathTracingMidpoint.simple_ipv6_packet(
+            eth_dst=ingress_router_mac,
+            eth_src=ptfadapter.dataplane.get_mac(0, ptf_port_idx),
+            ipv6_dst='2001:db8:2::1',
+            ipv6_src='2001:db8:1::1',
+            ipv6_hlim=64,
+        )
+
+    def ipv6_packet_with_hbh_pt(self, common_param, ptfadapter):
+        """ create IPv6 packet followed by a Hop-by-Hop Path Tracing Option with an empty MCD stack for testing """
+
+        (peer_ip_ifaces_pair, rif_rx_ifaces, rif_support, ptf_port_idx,
+         pc_ports_map, ptf_indices, ingress_router_mac) = common_param
+
+        ipv6_pkt = TestPathTracingMidpoint.simple_ipv6_packet(
```

Comment:
Is it possible to update this V6 extension to PTF utils repo? Will be useful for other test cases as well.